UIKitTests/UIApplication+NSNotificationTests.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UIApplication+NSNotificationTests.swift
+//  UIKitTests
+//
+//  Created by Chetan Agarwal on 30.11.18.
+//  Copyright © 2018 flowkey. All rights reserved.
+//
+
+import XCTest
+@testable import UIKit
+
+class UIApplication_NSNotificationTests: XCTestCase {
+    func testApplicationPostsWillEnterForegroundNotification() {
+        let notificationExpectation = expectation(forNotification: .UIApplicationWillEnterForeground,
```

Comment:
Cool that you wrote tests for this 👍 

Off the top of my head I'm unsure whether these will somehow call `UIApplication.shared` and/or `UIScreen.main` to be created - it would be best if we could avoid that because there's been no need to init our OpenGL rendering context in tests until now - it's costly, performance-wise, and reduces the number of environments we can run the tests in.

Would be great if you could investigate whether that's happening and if so we can find a way to avoid it (we have the ability to make a mock `UIScreen`, for example).

Reply:
@ephemer I checked this and it doesn't create an instance of `UIApplication`, and also the tests aways implicitly initialize the `UIScreen.main` with a mock `UIScreen` via `UIKitTests\TestSetup.swift`.

It's also why I post the notification using:
`NotificationCenter.default.post(name: name, object: UIApplication.shared)`
since `object` is an optional type  (`object anyObject: Any?`).

Philosophically, this is a kind of a catch-22 - since there are `UIApplication` lifecycle notifications being sent without an actual instance of the application, but I think, this should be acceptable. What are your thoughts about that?